src/sentry/api/endpoints/organization_slugs.py

```diff
@@ -40,31 +49,38 @@ def put(self, request: Request, organization) -> Response:
         rv = {}
 
         with transaction.atomic(router.db_for_write(Project)):
-            projects = {}
+            try:
+                projects = {}
 
-            # Clear out all slugs first so that we can move them
-            # around through the uniqueness
-            for project in project_q:
-                projects[str(project.id)] = project
-                project.slug = None
-                project.save()
+                # Clear out all slugs first so that we can move them
+                # around through the uniqueness
+                for project in project_q:
+                    projects[str(project.id)] = project
+                    project.slug = None
+                    project.save()
 
-            # Set new ones
-            for project_id, slug in slugs.items():
-                project = projects.get(project_id)
-                if project is None:
-                    continue
-                other = (
-                    Project.objects.filter(slug=slug, organization=organization)
-                    .exclude(id=project.id)
-                    .first()
+                # Set new ones
+                for project_id, slug in slugs.items():
+                    project = projects.get(project_id)
+                    if project is None:
+                        continue
+                    other = (
+                        Project.objects.filter(slug=slug, organization=organization)
+                        .exclude(id=project.id)
+                        .first()
+                    )
+                    if other is not None:
+                        if len(slugs) != len(slugs.values()):
+                            return Response({"detail": "Duplicate slug %s." % slug}, status=400)
+                    project.slug = slug
+                    project.update_option("sentry:reviewed-slug", True)
+                    project.save()
+                    rv[project_id] = slug
+            except (IntegrityError, MaxSnowflakeRetryError):
```

Comment:
Add try except here to catch duplicate slugs. I noticed when testing the check inside the try fails sometime so added this to be safe. There are no other code changes except better err messages.

Reply:
Do all IntegrityErrors and MaxSnowflakeRetryErrors mean duplicate slug? If not we should return a more generic error message.